### PR TITLE
Move jobframework constants into the directory

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

To make the jobframework stuff contained within the package. See related comment: https://github.com/kubernetes-sigs/kueue/pull/633#discussion_r1137383502

#### Which issue(s) this PR fixes:

Part of #369

#### Special notes for your reviewer:

The solution is to extract from the monolitic `testing` package the packages for job framework testing (batchv1.Job and MPIJob). Only these packages will depend on the `jobframwork` package.

### CODE REVIEW

File: pkg/controller/jobframework/constants.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```
Comment (alculquicondor) [inline pkg/controller/jobframework/constants.go:2]: 2023

File: pkg/controller/jobframework/reconciler.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	// when manageJobsWithoutQueueName is disabled we only reconcile jobs that have either
 	// queue-name or the parent-workload annotation set.
 	if !r.manageJobsWithoutQueueName && job.QueueName() == "" && isStandaloneJob {
-		log.V(3).Info(fmt.Sprintf("Neither %s, nor %s annotation is set, ignoring the job", constants.QueueAnnotation, constants.ParentWorkloadAnnotation))
+		log.V(3).Info(fmt.Sprintf("Neither %s, nor %s annotation is set, ignoring the job", QueueAnnotation, ParentWorkloadAnnotation))
```
Comment (alculquicondor) [inline pkg/controller/jobframework/reconciler.go:108]:
```suggestion
		log.V(3).Info(fmt.Sprintf("Neither %s label, nor %s annotation is set, ignoring the job", QueueLabel, ParentWorkloadAnnotation))
```

File: pkg/controller/jobframework/validation.go [hunk 3]
```diff
@@ -19,14 +19,12 @@ import (
...
-
-	"sigs.k8s.io/kueue/pkg/constants"
 )
 
 var (
 	annotationsPath       = field.NewPath("metadata", "annotations")
-	ParentWorkloadKeyPath = annotationsPath.Key(constants.ParentWorkloadAnnotation)
-	QueueNamePath         = annotationsPath.Key(constants.QueueAnnotation)
+	ParentWorkloadKeyPath = annotationsPath.Key(ParentWorkloadAnnotation)
+	QueueNamePath         = annotationsPath.Key(QueueAnnotation)
```
Comment (alculquicondor) [inline pkg/controller/jobframework/validation.go:27]: oh, we should validate the Label as well. I forgot to do this as part of the migration to v1beta1.

File: pkg/controller/jobs/job/job_controller.go [hunk 4]
```diff
@@ -118,11 +117,11 @@ func (j *Job) Object() client.Object {
 }
 
 func (j *Job) ParentWorkloadName() string {
-	return j.Annotations[constants.ParentWorkloadAnnotation]
+	return j.Annotations[jobframework.ParentWorkloadAnnotation]
 }
 
 func (j *Job) QueueName() string {
-	return j.Annotations[constants.QueueAnnotation]
```
Comment (alculquicondor) [inline pkg/controller/jobs/job/job_controller.go:125]: I think this was lost over the rebases.... We should check first in Labels, and, if not found, as Annotation.

File: pkg/controller/jobs/mpijob/mpijob_controller.go [hunk 5]
```diff
@@ -64,11 +63,11 @@ func (j *MPIJob) Object() client.Object {
 }
 
 func (j *MPIJob) ParentWorkloadName() string {
-	return j.Annotations[constants.ParentWorkloadAnnotation]
+	return j.Annotations[jobframework.ParentWorkloadAnnotation]
 }
 
 func (j *MPIJob) QueueName() string {
-	return j.Annotations[constants.QueueAnnotation]
+	return j.Annotations[jobframework.QueueAnnotation]
```
Comment (alculquicondor) [inline pkg/controller/jobs/mpijob/mpijob_controller.go:70]: same change here

File: pkg/util/testingjobs/job/wrappers.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+// PriorityClass updates job priorityclass.
+func (j *JobWrapper) PriorityClass(pc string) *JobWrapper {
+	j.Spec.Template.Spec.PriorityClassName = pc
+	return j
+}
+
+// Queue updates the queue name of the job
+func (j *JobWrapper) Queue(queue string) *JobWrapper {
+	j.Annotations[jobframework.QueueAnnotation] = queue
```
Comment (alculquicondor) [inline pkg/util/testingjobs/job/wrappers.go:86]: label

File: pkg/controller/jobframework/constants.go (hunk 1 above)
Comment (mimowo) [reply pkg/controller/jobframework/constants.go]: Done

File: pkg/controller/jobs/job/job_controller.go (hunk 4 above)
Comment (mimowo) [reply pkg/controller/jobs/job/job_controller.go:125]: Possibly, but if so, there wasn't a test for it, I'm going to add the support and the test, but I would like to move the function out from the interface as discussed here: https://github.com/kubernetes-sigs/kueue/pull/640#discussion_r1140250234.

File: pkg/util/testingjobs/job/wrappers.go (hunk 6 above)
Comment (mimowo) [reply pkg/util/testingjobs/job/wrappers.go]: done

File: pkg/controller/jobs/mpijob/mpijob_controller.go (hunk 5 above)
Comment (mimowo) [reply pkg/controller/jobs/mpijob/mpijob_controller.go]: Fixed

**Comment by mimowo** [comment]
@alculquicondor I think I have addressed the comments and simplified the interface so that the handling of labels and annotations for queue name is commonized. If we think MPIJob should support the spec.runPolicy.schedulingPolicy.Queue I will add this under a dedicated PR. This didn't work anyway, so let's progress the interface as needed.

File: pkg/controller/jobframework/validation.go [hunk 7]
```diff
@@ -19,16 +19,22 @@ import (
...
 )
 
+func ValidateCreateForQueueName(job GenericJob) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, ValidateLabelAsCRDName(job, QueueLabel)...)
+	allErrs = append(allErrs, ValidateAnnotationAsCRDName(job, QueueAnnotation)...)
+	return allErrs
+}
+
 func ValidateAnnotationAsCRDName(job GenericJob, crdNameAnnotation string) field.ErrorList {
```
Comment (alculquicondor) [inline pkg/controller/jobframework/validation.go:38]: can be unexported now?

File: test/integration/controller/job/job_controller_test.go [hunk 8]
```diff
@@ -243,6 +243,17 @@ var _ = ginkgo.Describe("Job controller", func() {
 				createdWorkload.Status.Conditions[0].Status == metav1.ConditionTrue
 		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
 	})
+	ginkgo.It("Should reconcile job when queueName set by annotation (deprecated)", func() {
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:246]: Thanks!

Add an empty line before `It`

File: pkg/controller/jobframework/validation.go (hunk 7 above)
Comment (mimowo) [reply pkg/controller/jobframework/validation.go:38]: Still is used by parent-workload, could follow up as for Queuename

File: test/integration/controller/job/job_controller_test.go (hunk 8 above)
Comment (mimowo) [reply test/integration/controller/job/job_controller_test.go:246]: Done

File: test/integration/controller/job/job_controller_test.go [hunk 9]
```diff
@@ -243,6 +243,17 @@ var _ = ginkgo.Describe("Job controller", func() {
...
+		jobQueueName := "test-queue"
+		job := testingjob.MakeJob(jobName, jobNamespace).QueueNameAnnotation("test-queue").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+		createdWorkload := &kueue.Workload{}
+		gomega.Eventually(func() error {
+			return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		gomega.Expect(createdWorkload.Spec.QueueName).Should(gomega.Equal(jobQueueName))
+	})
 	ginkgo.When("The parent-workload annotation is used", func() {
```
[the remark this answers is not in the document]
Comment (mimowo) [reply test/integration/controller/job/job_controller_test.go:259]: Done

File: pkg/controller/jobframework/validation.go [hunk 10]
```diff
@@ -39,19 +45,29 @@ func ValidateAnnotationAsCRDName(job GenericJob, crdNameAnnotation string) field
 	return allErrs
 }
 
+func ValidateLabelAsCRDName(job GenericJob, crdNameLabel string) field.ErrorList {
```
[the remark this answers is not in the document]
Comment (mimowo) [reply pkg/controller/jobframework/validation.go]: Yes, lowercased

**Pull request merged**